covariancematrix.go

```diff
@@ -80,3 +80,75 @@ func CovarianceMatrix(cov *mat64.Dense, x mat64.Matrix, wts []float64) *mat64.De
 	cov.Scale(1/(n-1), cov)
 	return cov
 }
+
+// CorrelationMatrix calculates a correlation matrix from a matrix of data,
+// using a two-pass algorithm. The matrix returned will be symmetric, square,
+// and positive-semidefinite.
+//
+// The weights wts should have the length equal to the number of rows in
+// input data matrix x. c should either be a square matrix with the same
```

Comment:
This should say something like 

````
If c is not nil, it must be a square matrix with the same number of columns as x, and the computed correlation matrix will be placed into c. If c is nil, a new Dense will be allocated and returned with the covariance matrix.
````
In short, add some text that the purpose of c is to be a receiver for the data.